docs/custom_modules.md

```diff
@@ -4,16 +4,74 @@
+The Drupal.org Contribution Tracker module (ct_drupal) automates tracking user contributions made specifically on the Drupal.org platform. It achieves this by:
 
-- [Contribution Plugin Manager](https://github.com/contrib-tracker/backend/tree/main/web/modules/custom/ct_manager): provides the feature to create a custom Plugin to track and store contributions from different sources and send notifications on Slack Channel.
+- The module utilizes a `ContributionSource` plugin named `DrupalContribution.php`. This plugin interacts with the `Drupal.org API` via a wrapper module built around `Guzzle 6` for efficient communication.
 
-- [Contribution Tracker Reports](https://github.com/contrib-tracker/backend/tree/main/web/modules/custom/ct_reports): provides functionality to generate reports based on contribution data.
+- To track contributions, the module requires each user's Drupal.org username to be filled in a dedicated field (managed by the `do_username` dependency).
 
-- [Contrib Tracker Users](https://github.com/contrib-tracker/backend/tree/main/web/modules/custom/ct_user): manages user-login activities in Contrib Tracker.
+- During cron runs, the system fetches contributions for all users with a populated Drupal.org username field. It retrieves the latest 100 code contributions and 100 issues for each user.
+
+- The plugin manager `ct_manager` plays the major role by invoking the plugin in `ct_drupal`. All the fetched data is passed backed to `ct_manager` to decide the storing process.
+
+### [GitHub Contribution Tracker](https://github.com/contrib-tracker/backend/tree/main/web/modules/custom/ct_github)
+
+This module automates tracking user contributions made on Github, specifically issues and code contributions. Here's how it works:
+
+- It utilizes a `ContributionSource` plugin named `GithubContribution.php` to interact with the Github GraphQL API.
+
+- This module requires each user's Github username to be filled in a dedicated field.
+
+- It requires a secure [GitHub personal access token](https://github.com/settings/tokens) stored in an environment variable or through other secure means and load it in settings.php.
+
+- During cron runs, the system fetches contributions for users with populated Github username fields. It retrieves the latest 100 issues and 100 code contributions for each user.
+
+- The plugin manager `ct_manager` plays the major role by invoking the plugin in `ct_drupal`. All the fetched data is passed backed to `ct_manager` to decide the storing process.
+
+In essence, this module simplifies tracking user activity on Github by offering an automated solution that integrates with existing Drupal functionalities and leverages the `ct_manager` module's infrastructure for processing and storage.
+
+### [Contribution Plugin Manager](https://github.com/contrib-tracker/backend/tree/main/web/modules/custom/ct_manager)
```

Comment:
I think we should have this module before `ct_drupal` and `ct_github`. The reason is that right now, you talk about `ContributionSource` plugin but haven't yet introduced the module that defines that plugin type. It feels a little confusing.

But if you start with this module, you can briefly mention that this plugin type is implemented by those modules. I think it will be more readable that way.